### PR TITLE
Added Dataverse plugin example

### PR DESCRIPTION
**Pull request by RajeevPentyala**
Provided plugin examples for Create and Update of 'AlmAcceleratorSamples' table. Update plugin contains usage of Images.

Fixes microsoft/coe-starter-kit#14

### CODE REVIEW

**Comment by filcole** [comment]
I think thi should be using a PlginBase class as a best practice example.  `pac plugin init` in v1.17.4 from the Power Platform CLI provides a very nice base plugin.

**Comment by RajeevPentyala** [comment]
> I think this should be using a PluginBase class as a best practice example. `pac plugin init` in v1.17.5 from the Power Platform CLI provides a very nice base plugin.
Thanks for the feedback @filcole. Updated the plugin example code with PluginBase.

File: ALMAcceleratorSampleSolution/Plugins/alm-plugins/PluginBase.cs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+            var localPluginContext = new LocalPluginContext(serviceProvider);
+
+            localPluginContext.Trace($"Entered {PluginClassName}.Execute() " +
+                $"Correlation Id: {localPluginContext.PluginExecutionContext.CorrelationId}, " +
+                $"Initiating User: {localPluginContext.PluginExecutionContext.InitiatingUserId}");
+
+            try
+            {
+                // Invoke the custom implementation
+                ExecuteCdsPlugin(localPluginContext);
```
Comment (filcole) [inline ALMAcceleratorSampleSolution/Plugins/alm-plugins/PluginBase.cs:58]: Should this be Dataverse rather than CDS?

Comment (RajeevPentyala) [reply ALMAcceleratorSampleSolution/Plugins/alm-plugins/PluginBase.cs]: @filcole PluginBase was auto generated file using pac. I agreed that name should be ExecuteDataversePlugin. Renamed the function.

Comment (filcole) [reply ALMAcceleratorSampleSolution/Plugins/alm-plugins/PluginBase.cs]: Thanks.  I logged that pac generates it incorrectly here: https://github.com/microsoft/powerplatform-vscode/issues/268

**Comment by devkeydet** [comment]
@RajeevPentyala - Let's remove the snk and use the new Plugin Package approach.  I would prefer we start with showcasing the new recommended way to author plugins moving forward.

**Comment by devkeydet** [comment]
Let's also consider putting some usage of a dependent assembly in here too.  It can be contrived.  Maybe something as simple as using json.net to parse some json and do something with it?  If you don't find something that fits in the sample app scenario, that's ok.  Just putt a comment in the code like "not doing anything meaningful here but wanted to showcase that you CAN reference third party assemblies in a plugin".

**Comment by devkeydet** [comment]
@filcole - Agree with your comments and THANK YOU!  Love that you are commenting on the PR and giving us feedback to make the example more meaningful and real world.

@RajeevPentyala  - In addition to the base class, I would like to see unit tests in this PR.  We want to come out the gate with the principle of showcasing not just "hello world", but real world recommendations.  I have an example here:
https://github.com/devkeydet/PrepareSelfContainedSolutionForExport/tree/main/src/SelfContainedSolutionTest/code-first

It has custom api example as well.  Note, that I had to make some changes to the base class to make it more unit test friendly.  Have a look at lets talk.  We should also provide feedback to the pac & VS extension teams about the base class that gets generated to make it more unit test friendly.

**Review by devkeydet — DISMISSED** [review]
See general comments on PR

**Comment by filcole** [comment]
Hi @devkeydet,

I understand the decision to not rely on non-MS unit testing frameworks (FakeXrmEasy, et. al.), I've found that a lot of the value of unit testing is the rapid iteration over a whole plugin.  However these unit testing frameworks often do not consider security, and may diverge from emulating the exact actual workings of a real Dataverse environment.

For these reasons I'd also like to see an example of an integration test for a plugin and custom API so that it can still be tested on a developer machine but the test would setup any data needed for the integration test, run the test, and then remove any data created.   This may be outside of the scope of this PR!

I wonder about demonstrating setting the insecure/secure configuration of a plugin during deployment, and an example of this configuration being accessed within the plugin.  However these days it may be preferable for the plugin to read and environment variable instead.

**Comment by devkeydet** [comment]
@filcole - Thanks for the feedback.  Your last comments are probably out of scope for this PR.  In fact, mine might be as well.  I'll let @mikefactorial and @RajeevPentyala determine what's in scope.  Depending on their feedback, we'll probably move what doesn't make it into this PR to separate enhancement issues.

**Pull request closed without merging**